code/modules/projectiles/guns/projectile/toy.dm

```diff
@@ -0,0 +1,102 @@
+/obj/item/weapon/gun/projectile/automatic/toy
+	name = "foam force SMG"
+	desc = "A prototype three-round burst toy submachine gun. Ages 8 and up."
+	icon = 'icons/obj/toyguns.dmi'
+	icon_state = "saber"
+	item_state = "gun"
+	mag_type = "/obj/item/ammo_box/magazine/toy/smg"
+	fire_sound = 'sound/items/syringeproj.ogg'
+	force = 0
+	throwforce = 0
+	burst_size = 3
+	can_suppress = 0
+	clumsy_check = 0
+	needs_permit = 0
+
+/obj/item/weapon/gun/projectile/automatic/toy/process_chambered()
+	return ..(0, 1)
+
+/obj/item/weapon/gun/projectile/automatic/toy/pistol
+	name = "foam force pistol"
+	desc = "A small, easily concealable toy handgun. Ages 8 and up."
+	icon_state = "pistol"
+	w_class = 2
+	mag_type = "/obj/item/ammo_box/magazine/toy/pistol"
+	can_suppress = 0
+	burst_size = 1
+	fire_delay = 0
+	action_button_name = null
+
+/obj/item/weapon/gun/projectile/automatic/toy/pistol/update_icon()
+	..()
+	icon_state = "[initial(icon_state)][chambered ? "" : "-e"]"
+
+/obj/item/weapon/gun/projectile/automatic/toy/pistol/riot
+	name = "foam force riot pistol"
+	desc = "RIOT! Ages 8 and up."
```

Comment:
In the `New()` proc for the pistol the mag is switched to the riot one. This makes the pistol start with a child of its standard mag, but still be able to chamber the parent mag's round.

I also changed a bit of code in `code/modules/projectiles/guns/projectile.dm` to make this work.